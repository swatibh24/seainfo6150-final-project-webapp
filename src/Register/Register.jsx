import React, {useEffect, useState} from "react";
import styles from "./Register.module.css";
import { Link } from "react-router-dom";

const Register = () => {

  const [submittedForm, setSubmittedForm] = useState();
  function onSubmit(e) {
    e.preventDefault();
    const data = new FormData(e.target);
    setSubmittedForm(data);
  };

  return (
    <div>
      {submittedForm ?  (<p className={styles.welcome}>Hi Welcome {submittedForm.get("username")}, Thank you for joining us!</p>) : (
          <form onSubmit={onSubmit} className={styles.containerBox}>
            <div className="logoBox">  
            <img
              className={styles.logo}
              src="http://shoppopbox.com/wp-content/uploads/2017/09/homechef-logo-square-1.jpg"/>
            </div>
            <h1 className={styles.registerTitle}>SignUp</h1>
            <div className={styles.usernameBox}>
              <input type="text" className={styles.input} name="username" id="username" placeholder="Please Enter Username *" required="required"/>
            </div>
            <div className={styles.passwordBox}>
              <input type="password" className={styles.input} name="password" id="password" placeholder="Please Enter Password *" required="required"/> <br/>
            </div>
            <div className={styles.emailBox}>
              <input type="email" className={styles.input} name="email" id="email" placeholder="Please Enter Email Address *" required="required"/><br/>
            </div>
            <div className={styles.zipcodeBox}>
              <input type="number" className={styles.opInput} name="zipcode" placeholder="Please Enter Zip Code" id="zipcode"/><br/>
            </div>
            <div className={styles.phoneBox}>
              <input type="number" className={styles.opInput} name="phone" placeholder="Please Enter Phone Number" id="phone"/> <br/>
            </div>
           
            <div className={styles.genderBox}>
              <label htmlFor="gender">Gender: </label>
              <select name="genderDropdown" id="genderDropdown" className={styles.dropdown}>
                <option value="male">Male</option>
                <option value="female">Female</option>
                <option value="decline">Decline to identify</option>
              </select>
            </div>
            <div className={styles.ageBox}>
                <label htmlFor="age">Age: </label>
                <select name="ageDropdown" id="ageDropdown" className={styles.dropdown}>
					<option value="1">1-10</option>
                    <option value="10">10-20</option>
                    <option value="20">20-30</option>
                    <option value="30">30-40</option>
                    <option value="40">40+</option>
                </select>
            </div>
            <div className={styles.checkBox}>
                <p>Interested in</p>
                <label htmlFor="soup" className={styles.checkItem}>Soup
                    <input type="checkbox" name="soup" className={styles.inputItem} />
                </label>
                <label htmlFor="appetizer" className={styles.checkItem}>Appetizer
                    <input type="checkbox" name="appetizer" className={styles.inputItem} />
                </label>
                <label htmlFor="mainCourse" className={styles.checkItem}>Main Course
                    <input type="checkbox" name="mainCourse" className={styles.inputItem} />
                </label>
				        <label htmlFor="snack" className={styles.checkItem}>Snack
                    <input type="checkbox" name="snack" className={styles.inputItem} />
                </label>
				        <label htmlFor="dessert" className={styles.checkItem}>Dessert
                    <input type="checkbox" name="dessert" className={styles.inputItem} />
                </label>
            </div>
            <input className={styles.register} type="submit" value="SignUp" />
          </form>
        )
      }
    </div>
  )
}

export default Register;