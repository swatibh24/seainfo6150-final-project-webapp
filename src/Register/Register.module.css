.containerBox{
    position: relative;
    z-index: 1;
    max-width: 380px;
    max-height: 800px;
    margin: 100px auto;
    display: block;
    width: 50%;
    padding: 40px 30px 10px 30px;
    background-image: linear-gradient(to right, #06719b, #5d80cc);
}

.logo{
    border-radius: 25%;
    width: 50px;
    height: 50px;
    display:block;
    margin:0 auto;
  }
.logoBox{
    text-align:center;
}

.registerTitle{
    color:white;
    text-align:center;
}
.checkBox{
margin-bottom:20px;
}
.input{
    border-radius: 50px !important;
    padding: 12px 15px 12px 15px !important;
    width: 100%;
    box-sizing: border-box;
    border: none !important;
    border: 1px solid #F3E5F5 !important;
    font-size: 12px !important;
    color: #000 !important;
    font-weight: 400
}

.opinput{
    background-color: white;
    border-radius: 50px !important;
    padding: 12px 15px 12px 15px !important;
    width: 100%;
    box-sizing: border-box;
    border: none !important;
    border: 1px solid #F3E5F5 !important;
    font-size: 12px !important;
    color: #000 !important;
    font-weight: 400
}


.usernameBox{
    margin-top:20px;
}
.passwordBox{
    margin-top:20px;
}
.emailBox{
    margin-top:20px;
}
.zipcodeBox{
    margin-top:10px;
}
.phoneBox{
    margin-top:10px;
}
.genderBox{
    margin-top:15px;
}
.ageBox{
    margin-top:15px;
}

.alert{
    font-weight: bold;
    margin: 100px 37%;
    color: orange
}
.welcome{
    margin: 15% 25%;
    font-size: 50px;
    font-weight: bold;
    color: grey;
    text-align:center;
}


.register{
    border-color: transparent;
    font-size:1rem;
    width:60%;
    margin-left:20%;
    cursor:pointer;
    margin-bottom: 50px;
    outline: none;
    text-decoration: none;
    margin-top: 10px;
    font-weight: 600;
    padding: 1rem 2rem;
    text-transform: uppercase;
    transition-property: all;
    transition-duration: 0.5s;
    box-shadow: 1px 1px 4px 0 rgba(0,0,0,.4);
    transition-timing-function: cubic-bezier(0.65,-0.25,0.25,1.95);
    border-radius: 25px;
}