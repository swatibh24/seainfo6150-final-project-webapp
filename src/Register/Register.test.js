import React from "react";
import { render } from "@testing-library/react";
import Register from "./Register.jsx";


describe("Register tests", () => {
    it("renders correctly", () => {
      const { container } = render(<Register />);
      expect(container).toMatchSnapshot();
    });
  });