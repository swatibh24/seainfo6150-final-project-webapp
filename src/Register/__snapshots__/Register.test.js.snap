// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register tests renders correctly 1`] = `
<div>
  <div>
    <form
      class="containerBox"
    >
      <div
        class="logoBox"
      >
        <img
          class="logo"
          src="http://shoppopbox.com/wp-content/uploads/2017/09/homechef-logo-square-1.jpg"
        />
      </div>
      <h1
        class="registerTitle"
      >
        SignUp
      </h1>
      <div
        class="usernameBox"
      >
        <input
          class="input"
          id="username"
          name="username"
          placeholder="Please Enter Username *"
          required=""
          type="text"
        />
      </div>
      <div
        class="passwordBox"
      >
        <input
          class="input"
          id="password"
          name="password"
          placeholder="Please Enter Password *"
          required=""
          type="password"
        />
         
        <br />
      </div>
      <div
        class="emailBox"
      >
        <input
          class="input"
          id="email"
          name="email"
          placeholder="Please Enter Email Address *"
          required=""
          type="email"
        />
        <br />
      </div>
      <div
        class="zipcodeBox"
      >
        <input
          class="opinput"
          id="zipcode"
          name="zipcode"
          placeholder="Please Enter Zip Code"
          type="number"
        />
        <br />
      </div>
      <div
        class="phoneBox"
      >
        <input
          class="opinput"
          id="phone"
          name="phone"
          placeholder="Please Enter Phone Number"
          type="number"
        />
         
        <br />
      </div>
      <div
        class="genderBox"
      >
        <label
          for="gender"
        >
          Gender: 
        </label>
        <select
          class="dropdown"
          id="genderDropdown"
          name="genderDropdown"
        >
          <option
            value="male"
          >
            Male
          </option>
          <option
            value="female"
          >
            Female
          </option>
          <option
            value="decline"
          >
            Decline to identify
          </option>
        </select>
      </div>
      <div
        class="ageBox"
      >
        <label
          for="age"
        >
          Age: 
        </label>
        <select
          class="dropdown"
          id="ageDropdown"
          name="ageDropdown"
        >
          <option
            value="1"
          >
            1-10
          </option>
          <option
            value="10"
          >
            10-20
          </option>
          <option
            value="20"
          >
            20-30
          </option>
          <option
            value="30"
          >
            30-40
          </option>
          <option
            value="40"
          >
            40+
          </option>
        </select>
      </div>
      <div
        class="checkBox"
      >
        <p>
          Interested in
        </p>
        <label
          class="checkItem"
          for="soup"
        >
          Soup
          <input
            class="inputItem"
            name="soup"
            type="checkbox"
          />
        </label>
        <label
          class="checkItem"
          for="appetizer"
        >
          Appetizer
          <input
            class="inputItem"
            name="appetizer"
            type="checkbox"
          />
        </label>
        <label
          class="checkItem"
          for="mainCourse"
        >
          Main Course
          <input
            class="inputItem"
            name="mainCourse"
            type="checkbox"
          />
        </label>
        <label
          class="checkItem"
          for="snack"
        >
          Snack
          <input
            class="inputItem"
            name="snack"
            type="checkbox"
          />
        </label>
        <label
          class="checkItem"
          for="dessert"
        >
          Dessert
          <input
            class="inputItem"
            name="dessert"
            type="checkbox"
          />
        </label>
      </div>
      <input
        class="register"
        type="submit"
        value="SignUp"
      />
    </form>
  </div>
</div>
`;
